gprofiler/main.py

```diff
@@ -730,7 +802,8 @@ def init_pid_file(pid_file: str) -> None:
 
 def main() -> None:
     args = parse_cmd_args()
-    verify_preconditions(args)
+    if args.subcommand is None:
```

Comment:
```suggestion
    if args.subcommand != "upload-file":
```

Let's be conservative, and be explicit for that case.